soroban-simulation/src/simulation.rs

```diff
@@ -333,11 +342,16 @@ impl SimulationAdjustmentConfig {
 fn create_transaction_data(
     resources: SorobanResources,
     resource_fee: i64,
+    new_entries_proof: Option<ArchivalProof>,
 ) -> SorobanTransactionData {
     SorobanTransactionData {
         resources,
         resource_fee,
-        ext: ExtensionPoint::V0,
+        ext: if let Some(proof) = new_entries_proof {
+            SorobanTransactionDataExt::V1(vec![proof].try_into().unwrap())
+        } else {
+            SorobanTransactionDataExt::V0
```

Comment:
This is fine for now, but in my prototype I've had all SorobanTransactionDataExt default to V1 regardless of if there's a proof or not in p23 (it's just an empty vector if no proof is required). I don't think this is strictly necessary and it does increase TX size overhead slightly, but I thought the status quo for XDR changes like this is to require the new version of the extension. I don't feel strongly about this either way, just wanted to flag it.

Reply:
Hmm actually this is not currently compatible with my prototype, where a soroban tx gets dropped and is not considered valid if we're in p23 and SorobanTransactionDataExt is not v1. I can either change my prototype or we can change it here.

Reply:
This is not up to core to decide, is it? I think we should keep V0 around for the sake of backwards compatibility. I think for tx envelope changes we should be as backwards compatible as possible and shouldn't deprecate any exts unless they're somehow broken/incompatible with the protocol (which isn't the case here).